src/Meilisearch/Index.cs

```diff
@@ -67,23 +79,49 @@ public async Task<bool> Delete()
         /// Add documents.
         /// </summary>
         /// <param name="documents">Documents to add.</param>
+        /// <param name="primaryKey">Primary key for the documents.</param>
         /// <typeparam name="T">Type of the document. Even though documents are schemaless in MeiliSearch, making it typed helps in compile time.</typeparam>
         /// <returns>Returns the updateID of this async operation.</returns>
-        public async Task<UpdateStatus> AddDocuments<T>(IEnumerable<T> documents)
+        public async Task<UpdateStatus> AddDocuments<T>(IEnumerable<T> documents, string primaryKey = default)
         {
-            var responseMessage = await this.client.PostAsJsonAsync($"/indexes/{this.Uid}/documents", documents);
+            HttpResponseMessage responseMessage;
+            string uri = $"/indexes/{this.Uid}/documents";
+            if (primaryKey == default)
+            {
+                responseMessage = await this.client.PostAsJsonAsync(uri, documents);
+            }
+            else
+            {
+                this.PrimaryKey = primaryKey;
+                uri = QueryHelpers.AddQueryString(uri, new { primaryKey = primaryKey }.AsDictionary());
+                responseMessage = await this.client.PostAsJsonAsync(uri, documents);
+            }
```

Comment:
We should not add the primary key as the `primaryKey` of the index (`this.PrimaryKey`) since the user can pass something that will not be taken into account.